base_layer/core/src/blocks/block.rs

```diff
@@ -62,7 +63,7 @@ pub struct Block {
 
 impl Block {
     // create a total_coinbase offset containing all fees for the validation
-    pub fn calculate_coinbase_and_fees(&self, rules: &ConsensusManager) -> MicroTari {
+    pub fn calculate_coinbase_and_fees<B: BlockchainBackend>(&self, rules: ConsensusManager<B>) -> MicroTari {
```

Comment:
This is starting to look ugly.

And now `Block` has to know about `BlockchainBackend`.

I'm thinking it's better to move this method into `ConsensusManager` with the following signature:
```
pub fn calculate_coinbase_and_fees(&self, block: &Block) -> MicroTari 
```

Backend dep be gone! And things sit more where they belong.


If private access into Block is an issue, then add a `total_fee()` method that sums the fees from the kernels as a separate function.

Reply:
Done